src/file-events/headers/generic_fsnotifier.h

```diff
@@ -178,6 +178,17 @@ class TerminateCommand : public Command {
     }
 };
 
+struct JniConstants {
+    JniConstants(JNIEnv* env);
+    void unload(JNIEnv* env);
+
+    const jclass nativeExceptionClass;
+    const jclass classClass;
+    const jclass nativeFileWatcherClass;
+};
+
+extern JniConstants* jniConstants;
```

Comment:
Why `extern` and not `static`? We don't want this to be accessible outside, no?

Reply:
If we declare it `static` in the header, then every object file will have the field compiled in. I don't think that is what we want. We can remove the declaration from the header completely, given that it is currently only used in `generic_fsnotifier.cpp`. If we want other `.cpp` files to use the `jniConstants`, then we need to use extern.

Reply:
OK. I'd remove it from the header until we actually need to export it, but I'm okay with this too.

Reply:
Actually, I might need these from `logging.cpp`, so let's keep it `extern`.